src/main/java/io/pravega/perf/PravegaPerfTest.java

```diff
@@ -44,36 +40,39 @@
  * Data format is in comma separated format as following: {TimeStamp, Sensor Id, Location, TempValue }.
  */
 public class PravegaPerfTest {
-
+    private static final int maxTime = 60 * 60 * 24;
     private static String controllerUri = "tcp://localhost:9090";
```

Comment:
All these need to be final.

Reply:
controllerUri  can change depending on the user input

Reply:
Then it cannot be static. Otherwise you have a race condition.

Reply:
no threads access this .. its default and one time assignment.

Reply:
You are correct. This means all of these members can be removed altogether. They are only ever used by the same method that assigned them. So they can be local variables.

Reply:
I have refactored the this complete file. please check.